apps/entity-service/src/entities/processors/project-report.processor.ts

```diff
@@ -0,0 +1,195 @@
+      builder.where({ projectId: { [Op.in]: ProjectUser.userProjectsSubquery(userId) } });
+    } else if (permissions?.includes("projects-manage")) {
+      builder.where({ projectId: { [Op.in]: ProjectUser.projectsManageSubquery(userId) } });
+    }
+
+    const associationFieldMap = {
+      organisationUuid: "$project.organisation.uuid$",
+      country: "$project.country$",
+      projectUuid: "$project.uuid$"
+    };
+
+    for (const term of [
+      "status",
+      "updateRequestStatus",
+      "frameworkKey",
+      "organisationUuid",
+      "country",
+      "projectUuid"
+    ]) {
+      if (query[term] != null) {
+        const field = associationFieldMap[term] || term;
+        builder.where({ [field]: query[term] });
+      }
+    }
+
+    if (query.search != null) {
+      builder.where({
+        [Op.or]: [
+          { "$project.name$": { [Op.like]: `%${query.search}%` } },
+          { "$project.organisation.name$": { [Op.like]: `%${query.search}%` } }
+        ]
+      });
+    }
+
+    if (query.projectUuid != null) {
+      const project = await Project.findOne({ where: { uuid: query.projectUuid }, attributes: ["id"] });
+      if (project == null) {
+        throw new BadRequestException(`Project with uuid ${query.projectUuid} not found`);
+      }
+      builder.where({ projectId: project.id });
+    }
+
+    return { models: await builder.execute(), paginationTotal: await builder.paginationTotal() };
+  }
+
+  async addFullDto(document: DocumentBuilder, projectReport: ProjectReport) {
+    const projectReportId = projectReport.id;
+    const taskId = projectReport.taskId;
+    const reportTitle = await this.getReportTitle(projectReport);
+    const siteReportsIdsTaks = ProjectReport.siteReportIdsTaksSubquery([taskId]);
+    const seedsPlantedCount = (await Seeding.visible().siteReports(siteReportsIdsTaks).sum("amount")) ?? 0;
+    const treesPlantedCount =
+      (await TreeSpecies.visible().collection("tree-planted").siteReports(siteReportsIdsTaks).sum("amount")) ?? 0;
+    const approvedSiteReportsFromTask = await SiteReport.approved()
+      .reportsTask(taskId)
+      .findAll({ attributes: ["id", "siteId", "numTreesRegenerating"] });
+    const regeneratedTreesCount = sumBy(approvedSiteReportsFromTask, "numTreesRegenerating");
+    const siteReportsCount = await SiteReport.reportsTask(taskId).count();
+    const nurseryReportsCount = await NurseryReport.reportsTask(taskId).count();
+    const migrated = !!projectReport.oldModel;
+    const seedlingsGrown = await this.getSeedlingsGrown(projectReport);
+    const siteReportsUnsubmittedIdsTask = await ProjectReport.siteReportsUnsubmittedIdsTaskSubquery([taskId]);
+    const nonTreeTotal = (await Seeding.visible().siteReports(siteReportsUnsubmittedIdsTask).sum("amount")) ?? 0;
+    const readableCompletionStatus = await this.getReadableCompletionStatus(projectReport.completion);
+    const createdByUser = projectReport.user ? projectReport.user : null;
```

Comment:
This ternary isn't doing much - there's no reason this can't be simply `const createdByUser = projectReport.user`. If you need explicitly null over undefined, use `?? null` instead.